portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -49,11 +84,22 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
   public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
     // Get the input from the form. and add to comments array
     String comment = getParameter(request, "text-input", "");
-    comments.add(comment);
+    String name = getParameter(request, "name-input", "");
+    String limit = getParameter(request, "limit-input", "");
+    long timestamp = System.currentTimeMillis();
+
+    Entity commentEntity = new Entity("Comment");
+    commentEntity.setProperty("comment", comment);
+    commentEntity.setProperty("name", name);
+    commentEntity.setProperty("timestamp", timestamp);
+
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    datastore.put(commentEntity);
+
     response.sendRedirect("/index.html");
   }
 
-  private String convertToJson(List<String> comments) {
+  private String convertToJson(List<Comment> comments) {
```

Comment:
Pulling this into its own function was a good way to hide boilerplate code and keep the main method clean. nice!